### PR TITLE
Use unix.SysctlUint32("hw.ncpuonline") on OpenBSD

### PR DESCRIPTION
**Pull request by tklauser**
Use SysctlUvmexp from golang.org/x/sys/unix to avoid having to simplify
the implementation of cpu.InfoWithContext

Signed-off-by: Tobias Klauser <tklauser@distanz.ch>

### CODE REVIEW

File: cpu/cpu_openbsd.go [hunk 1]
```diff
@@ -169,19 +168,10 @@ func InfoWithContext(ctx context.Context) ([]InfoStat, error) {
...
-	buf, _, err := common.CallSyscall(mib)
-	if err != nil {
-		return nil, err
-	}
-
-	var ncpu int32
-	br := bytes.NewReader(buf)
-	err = binary.Read(br, binary.LittleEndian, &ncpu)
-	if err != nil {
+	if u32, err = unix.SysctlUint32("hw.ncpuonline"); err != nil {
```
Comment (Lomanic) [inline cpu/cpu_openbsd.go:171]: could you please rename `u32` to `ncpu`?

**Review by Lomanic — CHANGES_REQUESTED** [review]

**Comment by Lomanic** [comment]
Besides my small nit, this is a great modification, didn't know about this function in x/sys/unix, thanks for this PR @tklauser!

File: cpu/cpu_openbsd.go (hunk 1 above)
Comment (tklauser) [reply cpu/cpu_openbsd.go:171]: The idea was to reuse the `var` already declared and used to extract `hw.cpuspeed`, but I agree having a separate, properly named `var ncpu` makes it easier to read. FWIW, I also changed it above and renamed the existing `u32` to `mhz`.

**Comment by tklauser** [comment]
Thanks @Lomanic, addressed the nit and also made the corresponding change when getting `unix.SysctlUint32("hw.cpuspeed")`. PTAL.

**Pull request merged**